Source/ACE/Entity/Player.cs

```diff
@@ -536,7 +536,8 @@ public QueuedGameAction ExaminationQueuePop()
         public void GrantXp(ulong amount)
         {
             // until we are max level we must make sure that we send
-            var chart = DatabaseManager.Charts.GetLevelingXpChart();
+            XpTable xpTable = XpTable.ReadFromDat();

```

Comment:
The XpTable class is already a "singleton" and caches the dat loading. (I learned that from you what I did the CharGen stuff) You had me questioning if this was working, but I threw a breakpoint in there to confirm and it is working as intended.